script/license_check.py

```diff
@@ -0,0 +1,51 @@
+#!/usr/bin/env python3
+import os
+
+# This script find all the projects inside certain folder and check if the have license like GPL
+
+def find_projects(vendor_folder):
+    # although most path should be host/owner/project
+    # there are projects using gopkg.in which result in path like gopkg.in
+    
+    # - TODO:limit level to 5
+    # - TODO: some project have vendor folder in scm, result in duplication, like testify
+    # - any folder with a license file is a project
+
+    projects = []
+    # use as a queue
+    q = [vendor_folder]
+    while len(q) > 0:
+        d = q.pop()
+        for entry in os.scandir(d):
+            if entry.is_file() and entry.name.lower() == 'license':
+                license = get_license(d + "/" + entry.name)
+                projects.append({'p':d, 'l':license})
+            if entry.is_dir():
+                q.insert(0, d + "/" + entry.name)
+    return projects
+
+# Common license files  https://github.com/github/choosealicense.com/tree/gh-pages/_licenses
+def get_license(license_file):
+    s = open(license_file,'r').read()
+    # TODO: switch case style?
```

Comment:
I end up using a dict and make license type as key, license keyword as value, though for most of them, the key and value are same